linux/iscsi.go

```diff
@@ -112,11 +112,23 @@ func RescanAndLoginToTarget(volume *model.Volume) (err error) {
 			}
 		}
 	}
-
-	targetSet, err := PerformDiscovery(volume.DiscoveryIP)
-	if err != nil {
-		log.Errorf("Unable to Perform Discovery with discoveryIp: %s. Error: %s", volume.DiscoveryIP, err.Error())
-		return fmt.Errorf("Unable to Perform Discovery with discoveryIp: %s. Error: %s", volume.DiscoveryIP, err.Error())
+	var targetSet model.IscsiTargets
+	if len(volume.DiscoveryIPs) == 0 && volume.DiscoveryIP == "" {
+		return fmt.Errorf("no discovery ip present to discover iSCSI Targets")
+	}
+	// give preference to slice of discovery IPs over a single discovery IP
+	if len(volume.DiscoveryIPs) > 0 {
```

Comment:
i think we  need to change PerformDiscovery() to just take an array, and find any one reachable IP from it and use that for discovery.

Reply:
sure will refactor this. any corner cases to consider?